core/src/main/java/com/datastax/oss/driver/internal/core/connection/ExponentialReconnectionPolicy.java

```diff
@@ -143,6 +147,21 @@ public void close() {
     @NonNull
     @Override
     public Optional<Duration> nextDelay(Optional<Throwable> throwable) {
+      if (throwable.isPresent() && throwable.get() instanceof PoolReconnectionException) {
```

Comment:
Logic from here and from `ConstantReconnectionPolicy.java` is the same. Can't we extract it to separate class and create `static` method for example: `checkIfThrowableIsFatal`?